streams/src/main/java/org/apache/kafka/streams/StreamsMetrics.java

```diff
@@ -17,26 +17,84 @@
+     * @param recordLevel The recording level (e.g., INFO or DEBUG) for this sensor.
      * @param operationName Name of the operation, could be get / put / delete / etc.
      * @param tags Additional tags of the sensor.
      * @return The added sensor.
      */
-    Sensor addLatencySensor(String scopeName, String entityName, String operationName, String... tags);
+    Sensor addLatencySensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, String... tags);
 
     /**
      * Record the given latency value of the sensor.
      */
     void recordLatency(Sensor sensor, long startNs, long endNs);
+
+    /**
+     * Add a throughput sensor. This is equivalent to adding a sensor with metrics rate.
+     *
+     * @param scopeName Name of the scope, could be the type of the state store, etc.
+     * @param entityName Name of the entity, could be the name of the state store instance, etc.
+     * @param recordLevel The recording level (e.g., INFO or DEBUG) for this sensor.
+     * @param operationName Name of the operation, could be get / put / delete / etc.
+     * @param tags Additional tags of the sensor.
+     * @return The added sensor.
+     */
+    Sensor addThroughputSensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, String... tags);
+
+
+    /**
+     * Generic sensor creation. Note that for most cases it is advisable to use {@link #addThroughputSensor(String, String, String, Sensor.RecordLevel, String...)}
+     * or {@link #addLatencySensor(String, String, String, Sensor.RecordLevel, String...)} to ensure metric name well-formedness and conformity with the rest
+     * of the streams code base.
+     * @param scopeName Name of the scope, could be the type of the state store, etc.
+     * @param entityName Name of the entity, could be the name of the state store instance, etc.
+     * @param recordLevel The recording level (e.g., INFO or DEBUG) for this sensor.
+     * @param operationName Name of the operation, could be get / put / delete / etc.
+     */
+    Sensor sensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel);
+
+    /**
+     * Same as previous constructor {@link #sensor(String, String, String, Sensor.RecordLevel, Sensor...)} sensor}, but takes a set of parents as well.
+     */
+    Sensor sensor(String scopeName, String entityName, String operationName, Sensor.RecordLevel recordLevel, Sensor... parents);
+
+    /**
+     * Helper function. Measure the latency of an action. This is equivalent to
+     * startTs = time.nanoseconds()
+     * action.run()
+     * endTs = time.nanoseconds()
+     * sensor.record(endTs - startTs)
+     * @param time      Time object.
+     * @param action    Action to run.
+     * @param sensor    Sensor to record value.
+     */
+    void measureLatencyNs(final Time time, final Runnable action, final Sensor sensor);
```

Comment:
The issue is that, StreamsMetrics is a public class so any public functions of this interface will be accessible to users as well. If we really want to add it to let users be able to use it, we need to think about how to clearly differentiate with `recordLatency`, when to use which, etc. Personally I'd rather not exposing it but only for internal usage, and let users to make their own optimizations if they want.